CONTRIBUTING.md

```diff
@@ -576,6 +575,60 @@ that are part of this project but not production code. The canonical example
 of this is `junit`.</dd>
 </dl>
 
+
+Reviewing and accepting your contribution
+-----------------------------------------
+
+We review every contribution carefully to ensure that the change is of high
+quality and fits well with the rest of the Elasticsearch codebase. If accepted,
+we will merge your change and usually take care of backporting it to
+appropriate branches ourselves.
+
+We really appreciate everyone who is interested in contributing to
+Elasticsearch and regret that we sometimes have to reject contributions even
+when they might appear to make genuine improvements to the system. Reviewing
+contributions can be a very time-consuming task, yet the review team is small
+and our time is very limited. In some cases the time we would need to spend on
+reviews would outweigh the benefits of a change by a significant margin.
+
+We strongly recommend you discuss your change in a Github issue before spending
+much time on its implementation. We may have to reject your contribution if it
+takes the wrong approach to solving a problem or if it solves a problem that
+does not need solving. We do not enjoy seeing time wasted like that.
+
+We often reject contributions due to the low quality of the submission since
+low-quality submissions tend to take unreasonable effort to review properly.
+Quality is rather subjective so it is hard to describe exactly how to avoid
+this, but there are some basic steps you can take to reduce the chances of
+rejection. Follow the guidelines listed above when preparing your changes. Your
+PR must at least pass the test suite and you must add tests that correspond
+with your changes too. The code must be formatted neatly and must not include
+unnecessary extra changes since these can consume a lot of extra effort during
+reviews.
+
+We sometimes reject contributions if we find ourselves performing a lot of
+review iterations without making significant progress. This usually indicates
+that there are a large number of small or superficial issues that we would
+normally expect to be addressed ahead of a review, and dealing with these
+leaves us with less time to consider the deeper implications of your changes.
+It is a good idea to review your contribution yourself or in your own team
+before asking us for a review. You may be surprised how many comments you can
+anticipate and address by taking a short break and then carefully looking over
+your changes again.
+
+We expect you to follow up on review comments reasonably promptly, but
+recognise that everyone has many priorities for their time and may not be able
+to respond for several days. We will understand if you find yourself without
+the time to complete your contribution, but please let us know that you have
+stopped working on it. We will try to send you a reminder if we haven't heard
+from you in a while, but may end up closing your PR if you do not respond for
+too long.
```

Comment:
I tried to write a 'conversely, understand that we may take a while to get around to reviews' but the guilt was too much for me :grin: Suggestions welcome